### DISCUSSION TITLE
i3 with Discord

### DISCUSSION THREAD

**Discussion by david072** [i3/i3#5652] [Q&A] [answered]
Hey all,

I am currently using i3 on Fedora and have installed Discord via flatpak. Now, I created a layout file that puts Discord and Spotify on top of each other in one of my workspaces. This works fine, however Discord creates a little floating window on startup which messes up the entire layouting.

As you can see in the video, Spotify is correctly swallowed by the lower node, however when I start Discord, it is somehow also moved down. Note, that I tried to fix this by force-disabling floating mode for Discord windows, but to no success.

![Mon Sep  4 12:46:53 PM CEST 2023](https://github.com/i3/i3/assets/66440035/5b35b2dd-e6fd-49f7-9298-a75840b32a7a)

![discord-issue](https://github.com/i3/i3/assets/66440035/6f4d2fc2-b74c-4039-854e-7459060d2f2f)

Does anyone know how to fix this and make Discord be swallowed by the correct node?

(you can check my dotfiles, including my i3 config [here](https://github.com/david072/dotfiles))

**Comment by orestisfl** [comment]
> This works fine, however Discord creates a little floating window on startup which messes up the entire layouting.

My guess is that the little floating window is swallowed by the placeholder container and then the main discord window is treated as a normal window.

I would try to get more specific criteria for these two windows to see if you can distinguish them. What's the output of `xprop` on the main discord window?

**Comment by david072** [comment, marked as answer]
Thanks for the quick response!

Yes, this seems to be the case. Thank you for bringing this up, I completely forgot about the other window properties...

For reference, this is the output of `xprop` for the floating window:
``` 
_NET_WM_DESKTOP(CARDINAL) = 2
_NET_WM_STATE(ATOM) = _NET_WM_STATE_FOCUSED
WM_STATE(WM_STATE):
		window state: Normal
		icon window: 0x0
_NET_WM_USER_TIME(CARDINAL) = 14177080
WM_NORMAL_HINTS(WM_SIZE_HINTS):
		program specified location: 2730, 785
		program specified minimum size: 300 by 350
		program specified maximum size: 300 by 350
_MOTIF_WM_HINTS(_MOTIF_WM_HINTS) = 0x2, 0x0, 0x0, 0x0, 0x0
XdndAware(ATOM) = BITMAP
_NET_WM_BYPASS_COMPOSITOR(CARDINAL) = 2
_GTK_HIDE_TITLEBAR_WHEN_MAXIMIZED(CARDINAL) = 1
WM_NAME(UTF8_STRING) = "Discord Updater"
_NET_WM_NAME(UTF8_STRING) = "Discord Updater"
WM_WINDOW_ROLE(STRING) = "browser-window"
WM_CLASS(STRING) = "discord", "discord"
_NET_WM_WINDOW_TYPE(ATOM) = _NET_WM_WINDOW_TYPE_NORMAL
_NET_WM_PID(CARDINAL) = 5
WM_CLIENT_MACHINE(STRING) = "fedora-david"
WM_PROTOCOLS(ATOM): protocols  WM_DELETE_WINDOW, _NET_WM_PING, _NET_WM_SYNC_REQUEST 
``` 

and this is the output for the main window:
```
_NET_WM_DESKTOP(CARDINAL) = 2
_NET_WM_STATE(ATOM) =
WM_STATE(WM_STATE):
		window state: Normal
		icon window: 0x0
_NET_WM_USER_TIME(CARDINAL) = 14179034
_NET_WM_ICON(CARDINAL) =
WM_NORMAL_HINTS(WM_SIZE_HINTS):
		program specified location: 3842, 18
		program specified minimum size: 940 by 475
		program specified maximum size: 2147483647 by 2147483647
_MOTIF_WM_HINTS(_MOTIF_WM_HINTS) = 0x2, 0x0, 0x1, 0x0, 0x0
XdndAware(ATOM) = BITMAP
_NET_WM_BYPASS_COMPOSITOR(CARDINAL) = 2
WM_NAME(UTF8_STRING) = "Friends - Discord"
_NET_WM_NAME(UTF8_STRING) = "Friends - Discord"
WM_WINDOW_ROLE(STRING) = "browser-window"
WM_CLASS(STRING) = "discord", "discord"
_NET_WM_WINDOW_TYPE(ATOM) = _NET_WM_WINDOW_TYPE_NORMAL
_NET_WM_PID(CARDINAL) = 5
WM_CLIENT_MACHINE(STRING) = "fedora-david"
WM_PROTOCOLS(ATOM): protocols  WM_DELETE_WINDOW, _NET_WM_PING, _NET_WM_SYNC_REQUEST
```

I just added a condition for the title to match for `.*Discord$`, which fixes the issue, i.e.:
![image](https://github.com/i3/i3/assets/66440035/902d02aa-28a2-43ba-8606-dbe07bb57867)
